force-app/main/default/classes/ListDonation.cls

```diff
@@ -0,0 +1,166 @@
+public with sharing class ListDonation {
+
+    static final String HARD_CREDIT = 'Hard Credit';
+    static final String SOFT_CREDIT = 'Soft Credit';
+    static final String PARTIAL_SOFT_CREDIT = 'Partial Soft Credit';
+
+    /** 
+    * @author Salesforce.org
+    * @date 2021
+    * @description Donation History inner class.
+    */
+    public class DonationHistory{
+        public Id id { get; set; }
+        public String name { get; set; }
+        public Boolean isPrimary { get; set; }
+        public Date closeDate { get; set; }
+        public String role { get; set; }
+        public String creditType { get; set; }
+        public Decimal amount { get; set; }
+        public String paymentMethod { get; set; }
+    }
+
+    /** 
+    * @author Salesforce.org
+    * @date 2021
+    * @description method to return Donation History list.
+    */
+    @AuraEnabled
+    public static List<DonationHistory> getDonationHistory(Id contactId){
+        List<OpportunityContactRole> donations = [SELECT Id, Opportunity.Name, Opportunity.CloseDate, Opportunity.Amount, Contact.Email, Contact.Name, Role, isPrimary FROM OpportunityContactRole WHERE Opportunity.StageName = 'Closed Won' AND ContactId =: contactId order by Opportunity.CloseDate desc Limit 50 ];
```

Comment:
This query should probably use `Opportunity.IsWon = TRUE` instead of `Opportunity.StageName = 'Closed Won'`, since there could be more than one stage in a customer's org that is set up to be a Closed/Won stage.

Reply:
Additionally, could we add line breaks to this and uppercase 'order by' and 'desc'?